xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/objects/classes/UsedValuesListQueryBuilder.java

```diff
@@ -111,19 +115,44 @@ public Query build(ListClass listClass) throws QueryException
     {
         String statement;
 
-        if (hibernateStore.getDatabaseProductName() == DatabaseProduct.ORACLE) {
-            statement = String.format("select %1$s, str(%1$s) as unfilterable1 "
-                + "from BaseObject as obj, %2$s "
-                + "where obj.className = :className and obj.name <> :templateName"
-                + " and prop.id.id = obj.id and prop.id.name = :propertyName "
-                + "order by unfilterable1 desc", getSelectColumnAndFromTable(listClass));
+        // On Oracle, StringListProperty uses a CLOB column which cannot be used in GROUP BY.
+        // We need a special query that doesn't use GROUP BY for this case.
+        boolean usesClobColumn = usesClobColumn(listClass);
+
+        String[] selectAndFrom = getSelectColumnAndFromTable(listClass);
+        if (usesClobColumn) {
+            // For CLOB columns on Oracle, we cannot use GROUP BY or DISTINCT on CLOB columns.
+            // Unfortunately, we also cannot use subqueries in select clauses in HQL in Hibernate < 6 - otherwise, we
+            // could possibly have used a partitioned ROW_NUMBER() over (PARTITION BY ...) approach.
+            // We use a "not exists" subquery to get distinct values instead by selecting only the first occurrence of
+            // each value by ordering by id and ensuring no prior id has the same value.
+            String selectColumn = selectAndFrom[0];
+            String fromTable = selectAndFrom[1];
+            String fromTableProp2 = fromTable.replace(" as prop", " as prop2");
+            String selectColumnProp2 = selectColumn.replace("prop.", "prop2.");
+
+            statement = String.format(
+                "select %1$s as stringValue, 1L as unfilterable0 "
+                    + "from BaseObject as obj, %2$s "
+                    + "where obj.className = :className "
+                    + "  and obj.name <> :templateName "
+                    + "  and prop.id.id = obj.id "
+                    + "  and prop.id.name = :propertyName "
+                    + "  and not exists ("
+                    + "    select 1 from %3$s "
+                    + "    where prop2.id.name = :propertyName "
+                    + "      and prop2.id.id < prop.id.id "
+                    + "      and FUNCTION('DBMS_LOB.COMPARE', %4$s, %1$s) = 0"
+                    + "  ) "
+                    + "order by obj.id",
+                selectColumn, fromTable, fromTableProp2, selectColumnProp2);
         } else {
-            statement = String.format("select %1$s, count(*) as unfilterable0 "
+            statement = String.format("select %1$s as stringValue, count(*) as unfilterable0 "
```

Comment:
not sure what does it bring? Or was just for debugging purpose?

Reply:
hmmmm, I would add a comment very close to the queries then: I feel it's very error prone for future maintenance to assume that variable name is needed for the TextQueryFilter to work